src/reducers/players.js

```diff
@@ -1,16 +1,25 @@
 import {ActionTypes} from '../constants';
 import player, {selectors as playerSelectors} from './player';
 
-const {INITIAL_STATE, TILE_SELECTED} = ActionTypes;
+const {INITIAL_STATE, RACK_TILE_SELECTED, BOARD_TILE_SELECTED, END_TURN} = ActionTypes;
 
 export default function players(state = {}, action) {
   switch (action.type) {
-    case TILE_SELECTED:
-      action.turn = parseInt(action.player);
+    case RACK_TILE_SELECTED:
       return Object.assign({}, state, {
         1: player(state[1], action),
         2: player(state[2], action)
       });
+    case BOARD_TILE_SELECTED:
+      return Object.assign({}, state, {
+        1: player(state[1], {...action, player: 1}),
+        2: player(state[2], {...action, player: 2})
+      });
+    case END_TURN:
```

Comment:
it's better to name this `TURN_ENDED` because an action answers the question "what happened in the system?"  it doesn't issue a "command" of telling the app what to do (`END_TURN`).